### PR TITLE
Added tests for generic workers

### PR DESCRIPTION
**Pull request by thecsw**
Added tests for generic workers by chaining workers together and giving them unique work, where we also see if all the expected outputs match as well.

Signed-off-by: Sandy <sandy@sandyuraz.com>

### CODE REVIEW

File: workers_test.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	results := make([]int, 0, 1001)
+	go func() {
+		for i := 0; i <= 1000; i++ {
+			inputs <- i
+		}
+		close(inputs)
+	}()
+	for ii := range ints {
+		results = append(results, ii)
+	}
```
Comment (matthewsanetra) [inline workers_test.go:66]: Does this automatically close the `ints` channel?

**Review by matthewsanetra — APPROVED** [review]
Left a question that might prompt you to commit a change, but otherwise looks good! Feel free to merge whenever you're ready.

File: workers_test.go (hunk 1 above)
Comment (matthewsanetra) [reply workers_test.go:66]: NVM, I've just realised that `GenericWorkers` spawns a goroutine that closes it when it's done.

Comment (thecsw) [reply workers_test.go:66]: Yeah, you just have to close the input and then the workers will close the returned outputs channels

**Pull request merged**